### PR TITLE
Add fetchJson

### PR DESCRIPTION
**Pull request by djhi**

### CODE REVIEW

File: packages/data-json-server/README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ This Data Provider fits REST APIs powered by [JSON Server](https://github.com/ty
 | `updateMany`       | `PUT http://my.api.url/posts/123`, `PUT http://my.api.url/posts/456`, `PUT http://my.api.url/posts/789` |
 | `delete`           | `DELETE http://my.api.url/posts/123`                                                                    |
 
-**Note**: The JSON Server REST Data Provider expects the API to include a `X-Total-Count` header in the response to `getList` and `getManyReference` calls. The value must be the total number of resources in the collection. This allows solid-admin to know how many pages of resources there are in total, and build the pagination controls.
+**Note**: The JSON Server REST Data Provider expects the API to include a `X-Total-Count` header in the response to `getList` and `getManyReference` calls. The value must be the total number of resources in the collection. This allows react-admin to know how many pages of resources there are in total, and build the pagination controls.
```
Comment (slax57) [inline packages/data-json-server/README.md]: need to revert change to this line

File: packages/data-json-server/README.md [hunk 2]
```diff
@@ -51,6 +51,50 @@ export const App = () => (
 );
 ```
 
+### Adding Custom Headers
+
+The provider function accepts an HTTP client function as second argument. By default, they use react-admin's `fetchUtils.fetchJson()` as HTTP client. It's similar to HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```
Comment (slax57) [inline packages/data-json-server/README.md]:
```suggestion
The provider function accepts an HTTP client function as second argument. By default, they use solid-admin's `fetchJson()` as HTTP client. It's similar to HTML5 `fetch()`, except it handles JSON decoding and HTTP error codes automatically.
```

There is no member named `fetchUtils` in solid-admin (only in react-admin)

File: packages/data-json-server/README.md [hunk 3]
```diff
@@ -51,6 +51,50 @@ export const App = () => (
...
+
+**Tip**: The most common usage of custom headers is for authentication. `fetchJson` has built-on support for the `Authorization` token header:
+
+```js
+const httpClient = (url, options = {}) => {
+    options.user = {
+        authenticated: true,
+        token: 'SRTRDFVESGNJYTUKTYTHRG'
+    };
+    return fetchUtils.fetchJson(url, options);
```
Comment (slax57) [inline packages/data-json-server/README.md]:
```suggestion
    return fetchJson(url, options);
```

**Review by slax57 — CHANGES_REQUESTED** [review]

File: packages/data-json-server/README.md (hunk 1 above)
Comment (djhi) [inline packages/data-json-server/README.md]:
```suggestion
**Note**: The JSON Server REST Data Provider expects the API to include a `X-Total-Count` header in the response to `getList` and `getManyReference` calls. The value must be the total number of resources in the collection. This allows solid-admin to know how many pages of resources there are in total, and build the pagination controls.
```

**Review by slax57 — APPROVED** [review]

**Pull request merged**